imports/plugins/core/accounts/server/no-meteor/mutations/setAccountProfileCurrency.js

```diff
@@ -0,0 +1,51 @@
+import SimpleSchema from "simpl-schema";
+import ReactionError from "@reactioncommerce/reaction-error";
+
+const inputSchema = new SimpleSchema({
+  currencyCode: String,
+  accountId: String
+});
+
+/**
+ * @name accounts/setAccountProfileCurrency
+ * @memberof Mutations/Accounts
+ * @summary Sets users profile currency
+ * @param {Object} context - GraphQL execution context
+ * @param {Object} input - Necessary input for mutation. See SimpleSchema.
+ * @param {String} input.currencyCode - currency symbol to add to user profile
+ * @param {String} [input.accountId] - optional decoded ID of account on which entry should be updated, for admin
+ * @returns {Promise<Object>} with updated address
+ */
+export default async function setAccountProfileCurrency(context, input) {
+  inputSchema.validate(input);
+  const { appEvents, collections, userHasPermission, userId: userIdFromContext } = context;
+  const { Accounts, Shops } = collections;
+  const { currencyCode, accountId: providedAccountId } = input;
+
+  const accountId = providedAccountId || userIdFromContext;
+  if (!accountId) throw new ReactionError("access-denied", "You must be logged in to set profile currency");
+
+  const account = await Accounts.findOne({ _id: accountId }, { fields: { shopId: 1 } });
+  if (!account) throw new ReactionError("not-found", "No account found");
+
+  if (userIdFromContext !== providedAccountId) {
```

Comment:
Might as well add the `!context.isInternalCall` check here like in the other one.